app/Ptr/PtrValidateRdnsTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Packages\Rdns\App\Ptr;
+
+use Packages\Rdns\App\RdnsTestCase;
+use Packages\Rdns\App\Ptr\PtrValidateRdns;
+use Packages\Rdns\App\Ptr\DnsRecordService;
+use Mockery;
+
+class PtrValidateRdnsTest extends RdnsTestCase {
+    
+    public function setUp() {
+        parent::setUp();
+        $this->dns = Mockery::mock(DnsRecordService::class);
+    }
+    /**
+     * @return PtrValidateRdns
+     */
+    private function rdns() {
+      return new PtrValidateRdns($this->dns);
+    }
+
+    public function testGivenValidIPItShouldReturnTrue(){
+        $ip = '1.1.1.1';
+        $this->dns
+          ->shouldReceive('get')
+          ->andReturn(
+            array(
+              array("host"=> "test-ptr","ip"=> '1.1.1.1', "type" => "A")              
+            )
+          );
+        $result = $this->rDns()->validate($ip, 'test-ptr');
+        $this->assertEquals($result, true);
+    }
+
+    public function testGivenValidIPV6ItShouldReturnTrue(){
+        $ip = "fe80:1:2:3:a:bad:1dea:dad";
+        $this->dns
+          ->shouldReceive('get')
+          ->andReturn(
+            array(
+              array("host"=> "test-ptr","ipv6"=> 'fe80:1:2:3:a:bad:1dea:dad', "type" => "AAAA")
+            )
+          );
+        $result = $this->rDns()->validate($ip, 'test-ptr');
+        $this->assertEquals($result, true);
+    }
+
+    public function testGivenInvalidIPItShouldReturnFalse(){
```

Comment:
these repetitive tests would be a great use case for a dataProvider. Not worth switching it now but it'd make these much shorter and, more importantly, easier to understand.